### PR TITLE
Pt 156157920 crash under sync recovery

### PR DESCRIPTION
**Pull request by ThomasArts**
When a worker that fetches blocks from another chain crashes unexpectetly, we should remve it from the pool of sync workers.

The difficult part of this patch is the test case.

### CODE REVIEW

File: apps/aecore/test/aecore_sync_SUITE.erl [hunk 1]
```diff
@@ -310,6 +312,54 @@ check_no_system_metrics_sent() ->
...
+
+    %% Set the same mining_rate to validate target
+    application:set_env(aecore, expected_mine_rate, 10), 
+
+    aecore_suite_utils:start_node(Dev2, Config),
+    aecore_suite_utils:connect(N2),
+    ct:log("node connected ~p", [N2]),
+    ok = rpc:call(N2, application, set_env, [aecore, ping_interval_limits, {500, 1000}], 5000),
+
+    spawn_link(fun() -> kill_sync_worker(N2, <<"http://ThomasComputer:3013">>) end),
```
Comment (hanssv) [inline apps/aecore/test/aecore_sync_SUITE.erl]: This looks interesting - either the node name doesn't matter or else something is fishy?

**Comment by hanssv** [comment]
I also have some slight concerns about the test being timing dependent, but it might be hard to avoid here...

File: apps/aecore/test/aecore_sync_SUITE.erl (hunk 1 above)
Comment (velzevur) [reply apps/aecore/test/aecore_sync_SUITE.erl]: It doesn't really matter - it is going to be killed right away

Comment (ThomasArts) [reply apps/aecore/test/aecore_sync_SUITE.erl]: It matters... only when we kill the process that is syncing, we test the right thing. Updated this.

**Pull request merged**